### ISSUE TITLE
2020/2/28

### ISSUE THREAD

**Issue by mingxoxo** [mingxoxo/deapLearning-study#3]
`
import numpy as np
import torch

t = np.array([0., 1., 2., 3., 4., 5., 6.]) #1D array 선언
print(t)

print("Rank of t: ", t.ndim) #차원
print("Shape of t", t.shape)  #차원안의 7개의 element

print('t[0] t[1] t[-1] = ', t[0], t[1], t[-1])
print('t[2:5] t[4:-1] = ', t[2:5], t[4:-1])
print('t[:2] t[3:] = ', t[:2], t[3:])
`

`
import numpy as np
import torch

t = torch.FloatTensor([0., 1., 2., 3., 4., 5., 6.])
print(t)

print(t.dim()) #차원
print(t.shape) #shape
print(t.size()) #shape

print('t[0] t[1] t[-1] = ', t[0], t[1], t[-1]) #element
print('t[2:5] t[4:-1] = ', t[2:5], t[4:-1]) #Slicing
print('t[:2] t[3:] = ', t[:2], t[3:]) #Slicing 
`

**Comment by mingxoxo** [comment]
**덧셈과 뺄셈을 할 때에는 두 tensor간의 크기가 같아야 한다.**
**행렬 곱을 수행할 때에도 마지막 차원과 첫번째 차원의 크기가 일치해야 한다**

파이토치에서 제공하는 Broadcasting
->크기가 맞지않는 것을 자동으로 맞춰서 계산해줌
(주의해서 사용해야 함! )
```
import torch

#Same shape
m1 = torch.FloatTensor([[3, 3]])
m2 = torch.FloatTensor([[2, 2]])
print(m1 + m2)

#Vector + scalar
m1 = torch.FloatTensor([[1, 2]])
m2 = torch.FloatTensor([[3]]) # 3 -> [[3, 3]] 같은 크기로 변환
print(m1 + m2)

# 2 x 1 Vector + 1 x 2 Vector
m1 = torch.FloatTensor([[1, 2]])
m2 = torch.FloatTensor([[3], [4]])
print(m1 + m2)
```

**Comment by mingxoxo** [comment]
![image](https://user-images.githubusercontent.com/44529556/75463625-6a9a0a00-59c9-11ea-9788-f8c54f7d3b7c.png)

![image](https://user-images.githubusercontent.com/44529556/75464193-2eb37480-59ca-11ea-959e-59e20a303403.png)
print(t.mean(dim = 0)) #dim=0을 없애라 가로줄 1개로 만듬
*sum은 mean과 유사

Argmax -> 가장 큰 값의 index 반환
![image](https://user-images.githubusercontent.com/44529556/75464666-dfba0f00-59ca-11ea-90d6-83c680d49bf7.png)
